src/box/lua/config/configdata.lua

```diff
@@ -197,10 +199,199 @@ function methods.bootstrap_leader_name(self)
     return self._bootstrap_leader_name
 end
 
+local function find_peer_name_by_uuid(peers, instance_uuid)
+    for name, peer in pairs(peers) do
+        local uuid = instance_config:get(peer.iconfig_def,
+            'database.instance_uuid')
+        if uuid == instance_uuid then
+            return name
+        end
+    end
+    return nil
+end
+
+function methods.peer_name_by_uuid(self, instance_uuid)
+    return find_peer_name_by_uuid(self._peers, instance_uuid)
+end
+
+local function find_saved_names(iconfig)
+    if type(box.cfg) == 'function' then
+        local snap_path = snapshot.get_path(iconfig)
+        -- Bootstrap is going to be done, no names are saved.
+        if snap_path == nil then
+            return nil
+        end
+
+        -- Read system spaces of snap file.
+        return snapshot.get_names(snap_path)
+    end
+
+    -- Box.cfg was already done. No sense in snapshot
+    -- reading, we can get all data from memory.
+    local peers = {}
+    for _, row in ipairs(box.space._cluster:select()) do
+        table.insert(peers, {
+            instance_uuid = row[2],
+            instance_name = row[3],
+        })
+    end
+
+    local replicaset_name, replicaset_uuid
+    if box.info.replicaset then
+        replicaset_name = box.info.replicaset.name
+        replicaset_uuid = box.info.replicaset.uuid
+    else
+        replicaset_name = nil
+        replicaset_uuid = box.info.cluster.uuid
+    end
+
+    return {
+        replicaset_name = replicaset_name,
+        replicaset_uuid = replicaset_uuid,
+        instance_name = box.info.name,
+        instance_uuid = box.info.uuid,
+        peers = peers,
+    }
+end
+
+-- Return a map, which shows, which instances doesn't have a name
+-- set, info about the current replicaset name is also included in map.
+function methods.missing_names(self)
+    local missing_names = {
+        peers = {},
+    }
+
+    local saved_names = find_saved_names(self._iconfig_def)
+    if saved_names == nil then
+        -- All names will be set during replicaset bootstrap.
+        return missing_names
+    end
+
+    -- Missing name of the current replicaset.
+    if saved_names.replicaset_name == nil then
+        missing_names[self._replicaset_name] = saved_names.replicaset_uuid
+    end
+
+    -- Missing names of all peers (including the current replica).
+    for _, saved_peer in ipairs(saved_names.peers) do
```

Comment:
As proposed by @Totktonada we should iterate over config, not over snapshot. This way we'll get alerts for all instances, which're in config, but which don't have name set in snapshot, passed uuid won't have any effect on alerts.

The logic now is not so trivial from the user side: alerts are triggered for the instances, which has instance_uuid set, but don't have name in snapshot.

Reply:
Done

Reply:
Confirmed. Closing the discussion.